### PR TITLE
Add initialize! method and uncomment a test

### PR DESCRIPTION
**Pull request by lbenet**

### CODE REVIEW

**Comment by lbenet** [comment]
Requested by @mforets

**Comment by mforets** [comment]
awesome, thanks @lbenet

File: src/validatedODEs.jl [hunk 1]
```diff
@@ -573,14 +589,14 @@ function validated_integ(f!, X0, t0::T, tmax::T, orderQ::Int, orderT::Int, absto
                                 nsteps, orderT, red_abstol, params, 
                                 parse_eqs, adaptive, minabstol, 
                                 absorb, check_property)
-        δtI = sign_tstep * Interval{T}(0, sign_tstep*δt)
+        δtI = sign_tstep * Interval(zt, sign_tstep*δt)
 
-        # New initial conditions and time
+        # New initial conditions and time, and output vectors
         nsteps += 1
+        @inbounds tv[nsteps] = t0
```
Comment (mforets) [inline src/validatedODEs.jl:596]: @lbenet was this change intended? note that `tb[nsteps]` came *after* updating `t0 += δt`. so the result is now different. seems like a breaking change (see https://github.com/JuliaReach/ReachabilityAnalysis.jl/pull/485/checks?check_run_id=2690783070#step:6:347)

Comment (mforets) [reply src/validatedODEs.jl:596]: to give a concrete example, consider:

```julia
@taylorize function f!(dx, x, p, t)
    dx[1] = x[2]
    dx[2] = zero(x[2])
    return dx
end

# Initial conditions
tini, tend = 0.0, 0.1
X0 = interval(0.95, 0.99) × interval(0.1, 0.1)

# Parameters
abstol = 1e-10
orderQ = 2
orderT = 4
ξ = set_variables("ξₓ ξᵥ", order=2*orderQ, numvars=length(q0))

tTM, qv, qTM = validated_integ(falling_ball!, X0, tini, tend, orderQ, orderT, abstol);

@show tTM;
@show domain(qTM[:, 1][1]);
@show domain(qTM[:, 2][1]);

tTM = [0.0, 0.0]
domain((qTM[:, 1])[1]) = Interval(0.0, 0.0)
domain((qTM[:, 2])[1]) = Interval(0.0, 0.1)
```
here i expected the second component of `tTM` to be 0.1.

**Comment by lbenet** [comment]
Yes, the change was intended; sorry if I did not tagged it as breaking. There was a mismatch in the indices of `tTM` and `qTM`: The domain of `qTM[:,n]` (n-th time step) is from `0` to `δt`, and time 0 in that domain corresponded to `tTM[n-1]`; now it corresponds to `tTM[n]`.

**Comment by lbenet** [comment]
I agree with [your comment](https://github.com/JuliaIntervals/TaylorModels.jl/pull/119#issuecomment-850479099) that it may be a bit confusing, but this only happens in the first and second entry in time. My point is that the very first entry corresponds to the initial condition (the domain is `Interval(0,0)`), while the second corresponds to the TaylorModel1, at t=0, which holds in the domain. 

Using your example above, note that `qTM[:,1]` has only zero-order terms in `t`, while `qTM[1,2]` has the local solution in `t`. Note also how the domains differ.
```julia
julia> tTM[1]
0.0

julia> qTM[1,1]
  0.97 + 0.020000000000000018 ξₓ + Interval(0.0, 0.0)

julia> domain.(qTM[1,1])
 Interval(0.0, 0.0)

julia> tTM[2]
0.0

julia> qTM[1,2]
  0.97 + 0.020000000000000018 ξₓ + ( 0.1) t + Interval(0.0, 0.0)

julia> domain.(qTM[1,2])
Interval(0.0, 0.1)
```

**Comment by mforets** [comment]
absolutely. it will not be difficult to adapt how we are constructing flowpipes in ReachabilityAnalysis.jl, [which happens here](https://github.com/JuliaReach/ReachabilityAnalysis.jl/blob/master/src/Algorithms/TMJets/TMJets21b/post.jl#L46), to the new behavior of `tTM`.  i just wanted to raise the point so that we are on the same page.

**Comment by lbenet** [comment]
With the new "convention", [this line](https://github.com/JuliaReach/ReachabilityAnalysis.jl/blob/b1412dac81351e7e0654beab32c4d49812de4e07/src/Algorithms/TMJets/TMJets21b/post.jl#L51) has to be adapted to something like
```julia
δt = TimeInterval(tv[i] + domain(xTM1v[1, i]))
```
and `i` can safely run over `eachindex(tv)`.

This is what is done in #117, e.g. [here](https://github.com/JuliaIntervals/TaylorModels.jl/pull/117/files#diff-996f9094bdfdab252020c391cbc05e95ad6f5cc8e6ffc0cf64b147ad34e7477aR75).
